istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -486,6 +486,11 @@ func applyYAML(client kube.CLIClient, yamlContent, ns string) error {
 	if err := client.ApplyYAMLFiles(ns, yamlFile); err != nil {
 		return fmt.Errorf("failed applying manifest %s: %v", yamlFile, err)
 	}
+
+	err = os.Remove(yamlFile)
```

Comment:
This doesn't seem correct to me. Why we remove the temp file only if ApplyYAMLFiles has no error?

Reply:
IIRC I didn't think creating a temp file and apply with `ApplyYAMLFiles` is a great approach, since some systems may not have the file writing permission.

Reply:
Good idea. This part of the code converts the yaml string to a file and then back to a string. It can expose a method to avoid this redundant operation